pkg/virtctl/console/console.go

```diff
@@ -209,82 +107,63 @@ func TerminalWebsocketCallback(ws *websocket.Conn, resp *http.Response, connErr
 		close(stopChan)
 	}()
 
-	// If there is no obvious connection error, set up the terminal
-	if connErr == nil {
-		state, err := terminal.MakeRaw(int(os.Stdin.Fd()))
-		if err != nil {
-			return fmt.Errorf("Make raw terminal failed: %s", err)
+	go func() {
+		defer close(readStop)
+		buf := make([]byte, 1024, 1024)
+		for {
+			// reading stdout from console
+			n, err := stdoutReader.Read(buf)
+			if err != nil && err != io.EOF {
+				return
+			}
+			if n == 0 && err == io.EOF {
+				return
+			}
+			// Writing to stdout
+			_, err = out.Write(buf[0:n])
+			if err == io.EOF {
+				return
+			}
 		}
-		defer terminal.Restore(int(os.Stdin.Fd()), state)
-		fmt.Fprint(os.Stderr, "Escape sequence is ^]")
-	}
-
-	return NewWebsocketCallback(os.Stdin, os.Stdout, stopChan)(ws, resp, connErr)
-}
-
-func RequestFromConfig(config *rest.Config, vm string, namespace string, device string) (*http.Request, error) {
+	}()
 
-	u, err := url.Parse(config.Host)
-	if err != nil {
-		return nil, err
-	}
+	go func() {
+		defer close(writeStop)
+		buf := make([]byte, 1024, 1024)
+		for {
+			// reading from stdin
+			n, err := in.Read(buf)
+			if err != nil && err != io.EOF {
+				return
+			}
+			if n == 0 && err == io.EOF {
+				return
+			}
 
-	switch u.Scheme {
-	case "https":
-		u.Scheme = "wss"
-	case "http":
-		u.Scheme = "ws"
-	default:
-		return nil, fmt.Errorf("Unsupported Protocol %s", u.Scheme)
-	}
+			// the escape sequence
+			if buf[0] == 29 {
+				return
+			}
+			// Writing out to the console connection
+			_, err = stdinWriter.Write(buf[0:n])
+			if err == io.EOF {
+				return
+			}
+		}
+	}()
 
-	u.Path = fmt.Sprintf("/apis/kubevirt.io/v1alpha1/namespaces/%s/virtualmachines/%s/console", namespace, vm)
```

Comment:
I've always preferred to use golang's `path.Join()` builtin. It saves you the trouble of sanitizing the input.

Reply:
yep, I forget about that package all the time. I think I've used join everywhere in the new code. This line was removed.